numba/np/arrayobj.py

```diff
@@ -5047,6 +5047,28 @@ def array_to_array(context, builder, fromty, toty, val):
     return val
 
 
+@lower_cast(types.Array, types.Float)
+@lower_cast(types.Array, types.Integer)
```

Comment:
What about complex floating point numbers?

Reply:
Done! Thanks for the review